examples/failing/UnusableTypeClassMethodSynonym.purs

```diff
@@ -0,0 +1,9 @@
+-- @shouldFailWith UnusableDeclaration
+module Main where
+
+type M x = forall a. a
+
+class C a where
+  -- after synonym expansion, the type doesn't actually contain an `a`
+  c :: M a
```

Comment:
In any case, I'll add a test case with a bound variable with the same name as a class argument.